meta.go

```diff
@@ -183,29 +183,71 @@ func (a *astTransformer) Transform(node *gast.Document, reader text.Reader, pc p
 	if meta == nil {
 		return
 	}
+	if meta.Kind == yaml.DocumentNode {
+		meta = meta.Content[0]
+	}
+	if meta.Kind != yaml.MappingNode {
+		// only mapping node is supported as root
+		return
+	}
+
 	table := east.NewTable()
 	alignments := []east.Alignment{}
-	for range meta {
+	for i := 1; i == len(meta.Content)%2; i++ {
 		alignments = append(alignments, east.AlignNone)
 	}
 	row := east.NewTableRow(alignments)
-	for _, item := range meta {
+	valueNodes := make([]*yaml.Node, 0, len(meta.Content)/2)
+	for i := 0; i < len(meta.Content); i = i + 2 {
+		keyNode := meta.Content[i]
+		valueNodes = append(valueNodes, meta.Content[i+1])
+
 		cell := east.NewTableCell()
-		cell.AppendChild(cell, gast.NewString([]byte(fmt.Sprintf("%v", item.Key))))
+		cell.AppendChild(cell, gast.NewString([]byte(valueNodeToString(keyNode))))
 		row.AppendChild(row, cell)
 	}
 	table.AppendChild(table, east.NewTableHeader(row))
 
 	row = east.NewTableRow(alignments)
-	for _, item := range meta {
+	for _, item := range valueNodes {
 		cell := east.NewTableCell()
-		cell.AppendChild(cell, gast.NewString([]byte(fmt.Sprintf("%v", item.Value))))
+		cell.AppendChild(cell, gast.NewString([]byte(valueNodeToString(item))))
 		row.AppendChild(row, cell)
 	}
 	table.AppendChild(table, row)
 	node.InsertBefore(node, node.FirstChild(), table)
 }
 
+func valueNodeToString(node *yaml.Node) string {
+	if node == nil {
+		return ""
+	}
+	switch node.Kind {
+	case yaml.SequenceNode:
+		val := make([]string, len(node.Content))
+		for i := range node.Content {
+			val[i] = valueNodeToString(node.Content[i])
+		}
+		return fmt.Sprintf("%v", val)
+
+	case yaml.MappingNode:
+		if (len(node.Content) % 2) != 0 {
+			return "<broken mapping node>"
+		}
+		val := make(map[string]string, len(node.Content)%2)
+		for i := len(node.Content); i > 1; i = i - 2 {
+			k := valueNodeToString(node.Content[i-2])
+			val[fmt.Sprint(k)] = valueNodeToString(node.Content[i-1])
+		}
+		return fmt.Sprintf("%v", val)
+
+	case yaml.ScalarNode:
+		return node.Value
+	}
+
+	return fmt.Sprintf("<do not support yaml node kind '%v'>", node.Kind)
```

Comment:
I'd like to change api fo this func so we can pass error obj upstream ... - just not done because like to get some ACK to change api of this lib more drastically

Reply:
@yuin can you ACK Y/N/need-more-discussion?